test/e2e/network/dns.go

```diff
@@ -67,12 +66,39 @@ var _ = SIGDescribe("DNS", func() {
 		validateDNSResults(f, pod, append(wheezyFileNames, jessieFileNames...))
 	})
 
+	It("should resolve DNS of partial qualified names for the cluster ", func() {
+		// All the names we need to be able to resolve.
+		// TODO: Spin up a separate test service and test that dns works for that service.
+		namesToResolve := []string{
+			"kubernetes.default",
+			"kubernetes.default.svc",
+			"kubernetes.default.svc.cluster",
```

Comment:
kubernetes.default.svc.cluster is not supposed to be resolvable. That would mean that ``local`` would also be a DNS suffix, which is not. Also, as you can see, you've removed only ``kubernetes.default`` and ``kubernetes.default.svc`` from ``should provide DNS for the cluster`` while you're adding 3 here.

This is causing the test to fail.